### PR TITLE
feat: Implement Guilds

### LINKED ISSUE #4: Implement Guilds

### PR DESCRIPTION
**Pull request by VincentRPS**
closes #4, more information & implementation details on our planning board.

### CODE REVIEW

File: derailedapi/database.py [hunk 1]
```diff
@@ -167,11 +267,61 @@ def verify_token(token: str | None):
...
         raise HTTPError(401, 'Signature on Authorization is Invalid')
 
 
+def objectify(data: dict[str, Any] | list[Any]) -> dict[str, Any] | list[Any]:
+    if isinstance(data, dict):
+        for k, v in data.items():
+            if (
+                isinstance(v, int)
+                and v > 2147483647
+                or isinstance(v, int)
```
Comment (ooliver1) [inline derailedapi/database.py]: An `int` will never be `'permissions'`

File: derailedapi/database.py [hunk 2]
```diff
@@ -167,11 +267,61 @@ def verify_token(token: str | None):
...
+                and k == 'permissions'
+            ):
+                data[k] = str(v)
+            elif isinstance(v, list):
+                new_value = []
```
Comment (ooliver1) [inline derailedapi/database.py]: This is never used, right?

File: derailedapi/database.py [hunk 3]
```diff
@@ -167,11 +267,61 @@ def verify_token(token: str | None):
...
+        elif rm_fields:
+            return User.objects(User.id == user_id).defer(rm_fields).get()
+        else:
+            return User.objects(User.id == user_id).get()
+
     except (itsdangerous.BadSignature):
         raise HTTPError(401, 'Signature on Authorization is Invalid')
 
 
+def objectify(data: dict[str, Any] | list[Any]) -> dict[str, Any] | list[Any]:
```
Comment (ooliver1) [inline derailedapi/database.py]: This should be a typevar if a passed dict returns a dict, and a passed list returns a list

**Review by ooliver1 — CHANGES_REQUESTED** [review]

File: derailedapi/database.py (hunk 1 above)
Comment (VincentRPS) [reply derailedapi/database.py]: the `k` isn't an int?

File: derailedapi/database.py (hunk 2 above)
Comment (VincentRPS) [reply derailedapi/database.py]: seems like I forgot to set this value

**Pull request merged**